src/evo/deterministicmns.cpp

```diff
@@ -1451,8 +1451,9 @@ bool CheckProRegTx(const CTransaction& tx, const CBlockIndex* pindexPrev, TxVali
         return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-protx-payload");
     }
 
-    if (auto maybe_err = ptx.IsTriviallyValid(llmq::utils::IsV19Active(pindexPrev)); maybe_err.did_err) {
-        return state.Invalid(maybe_err.reason, std::string(maybe_err.error_str));
+    if (!ptx.IsTriviallyValid(llmq::utils::IsV19Active(pindexPrev), state)) {
```

Comment:
can we not mix backport and refactoring like this?

Reply:
agree, 05757e6 is not needed PRs backported next to it, should be moved to a separate PR